tools/pipelines/test-perf-benchmarks-using-built-artifacts.yml

```diff
@@ -0,0 +1,376 @@
+            displayName: Download test package
+            inputs:
+              source: specific
+              project: internal
+              pipeline: ${{ variables.artifactPipeline }}
+              preferTriggeringPipeline: true
+              allowPartiallySucceededBuilds: true
+              runVersion: latestFromBranch
+              runBranch: $(Build.SourceBranch)
+              artifact: pack
+              patterns: "**/${{ replace(replace(execTestPackage, '@', '' ), '/', '-') }}-*.tgz"
+              path: $(Pipeline.Workspace)/client/pack
+
+          # Install test package
+          - task: Bash@3
+            displayName: Install Test Package
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}
+              script: |
+                echo "Install ${{ execTestPackage }}"
+                TEST_PACKAGE_FILE_PATTERN=${{ replace(replace(execTestPackage, '@', '' ), '/', '-') }}-*.tgz
+                TEST_PACKAGE_PATH_PATTERN=$(Pipeline.Workspace)/client/pack/scoped/$TEST_PACKAGE_FILE_PATTERN
+
+                echo $TEST_PACKAGE_FILE_PATTERN
+                echo $TEST_PACKAGE_PATH_PATTERN
+
+                echo `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l`
+                echo `ls $TEST_PACKAGE_PATH_PATTERN`
+
+                # if [[ `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l` -eq 1 ]]; then
+                TEST_PACKAGE_TGZ=`ls $TEST_PACKAGE_PATH_PATTERN`
+                # else
+                #   ls -1 $TEST_PACKAGE_PATH_PATTERN
+                #   echo "##vso[task.logissue type=error]Test package '${{ execTestPackage }}' not found, or there are more then one found"
+                # fi
+
+                npm install $TEST_PACKAGE_TGZ
+
+          # Unpack test files
+          - task: Bash@3
+            displayName: Unpack test files
+            inputs:
+              targetType: 'inline'
+              script: |
+                echo "Unpack test files for ${{ execTestPackage }}"
+                TAR_FILE=${{ replace(replace(replace(replace(execTestPackage, '@fluidframework/', '' ), '@fluid-internal/', '' ),'@fluid-', '' ), '/', '-') }}
+                mkdir -p ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}/src/test
+                tar -xvf $(Pipeline.Workspace)/test-files/$TAR_FILE.test-files.tar -C $(Pipeline.Workspace)/test-files
+                mv $(Pipeline.Workspace)/test-files/test/* ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}/src/test
+
+          - task: Bash@3
+            displayName: 'Run execution-time tests'
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ execTestPackage }}
+              script: |
+                echo "Run execution-time tests for ${{ execTestPackage }}"
+                cp ${{ variables.testWorkspace }}/.npmrc . ;
+                npm i;
+                npm run build:test;
```

Comment:
I am not opposed to it per se, but why you decide to pack the source and build it here?  Just curious.

Reply:
Ah because of what I mention above in the original post:

> 1. For both experimental-tree and tree: The packed dist/test folder does not have the json files necessary to run the benchmark tests so we have to change the postpack script to copy over the .ts files from src/test and publish them as test_files, and in the pipeline, we have to run a separate npm build:test to build the test files before running the benchmark tests. See: https://github.com/microsoft/FluidFramework/commit/996ba23

There are a number of .json files required to run the tests that don't get built into `dist` essentially so we use this workaround. This is one of the changes @alexvy86  wanted your input on-- is it okay to use this pattern?

Reply:
Yeah, Kajari and I talked a bit about it... it doesn't feel ideal in that it needs to build the test source files again in the "consuming pipelines", but if we have tests that depend on other collateral under `src/` (json files, mocharc.js configuration, etc), we need to pack `src/test`. Now, maybe we could pack *both* `src/test` and `dist/test` and unzip and overlay both, if we wanted to avoid building the test files again?

Or make it so all our packages move all the necessary collateral to dist/test and consume it from there. Might need changes to policy-check so that we can have `build`/`build:test` scripts that do a bit more than just the current commands.

Reply:
+1 for just packing and overlaying both.

Reply:
Done!